### ISSUE TITLE
Allow multiple stage workflows where hosts can follow different paths (was: Unified workflow)

### ISSUE THREAD

**Issue by planeturban** [ansible/awx#6793]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
# Unified workflow

**TL;DR**: Possibility to concatenate multiple templates, treating them as one unit. Plus adding support for error handling on per host basis in a workflow. 

## Problem

 Todays workflow job templates (WFJT) are not usable in a large scale deployment environment with multiple teams writing their own playbooks for installing different platforms

Apart from being awesome it'd make AWX a better contender to other automation platforms **cough***vRA***cough** as engine for deploying virtual machines.

## Example
Run a multiple stage vm deployment (with a cleanup node for failed hosts) against an inventory of new hosts. If one of these hosts fails in a node the full inventory will be sent to the fail link of that node and even if the desired outcome is run the fail node on all hosts, the node will exit on a success and not mark the WFJT as failed.

- Team 1 are in charge of creating virtual machines, they create the playbooks "Clone", "Destroy" and "Add disk".
- Team 2 are in charge of server tuning and baseline installations, they create the playbooks "Server tuning", "Add interface", "Create and mount disk" contained in "New server installation". 
- Team 3 are the network guys. They create the playbooks "Get IP from IPAM" and "Delete from IPAM". 
- Team 4 are consumers of the playbooks above. Using a UWFT they create a simple vm with one root disk and two interfaces to later on add their own platform on top of it.
- Team 5 adds their application on.

````
Team 4: New server installation 
Team 1:  |       |---> Clone
Team 2:  |              |-----------------> Add interface (eth0)
Team 3:  |                                        |------------> Get IP from IPAM.
Team 2:  |-----------------> Server tuning.
Team 4:  |-----------------> Add interface (eth1).
Team 3:  |                         |------------> Get IP from IPAM.
Team 4:  |-----------------> Template: run `package` module
         |
         |---> <All above successful?>
         |       - Yes: ------------------------|
         |       - No:                          |
         |         | ----> Notify team 2 and 4. |
         |         | -----> Destoy machine.     |
         |                  |------> Fail host. |
Team 5:  |                                      | -> Install application
         |                                              |--> <OK?>
         |                                                    - No: Notify team 4 and 5, keep host.
         |---> Notify team 5.

````

In a single-host inventory this would not pose a problem since all (one) hosts should fail. But if one were to deploy a multi server app were there are more consumers such as `team 4`, one failed host would destroy all other hosts and they all would have to be rebuilt because (for instance) something as as simple as a upstream webserver being unavailable. 

## Propsal

* UWFT should override the inventory set in template **or** only work on templates set to blank inventory (and thus checked 'Prompt on launch').
* UWFT should inherit surveys from workflow/templates.
* Variables from one survey should be able to be mapped in a subsequent survey variable. 
* UFWT nodes links should have at least two (optimal three) selections: "Only failed", "Only successful" (, and "Only changed"), making it possible to do cleanups and streamlining workflows.
* New node type: Notification. Use a existing notification in AWX as a node. 
* New node type: Fail. Used when any type node has run successfully to fail the host and/or (workflow) template. Failed host can be removed from inventory. For instance: delete and cleanup dns from failed host ran ok -> fail host, do not include in following nodes.

**Comment by wenottingham** [comment]
Updated title to aid searching.
